src/Accounts/Accounts/Rest/InvokeAzRestCommand.cs

```diff
@@ -0,0 +1,169 @@
+
+        [Parameter(Mandatory = false, HelpMessage = "JSON format payload")]
+        [ValidateNotNullOrEmpty]
+        public string Payload { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        #endregion
+
+        IAzureContext context;
+        private IAzureRestClient _client;
+        private IAzureRestClient ServiceClient
+        {
+            get
+            {
+                if (_client == null)
+                {
+                    var clientFactory = AzureSession.Instance.ClientFactory;
+                    _client = clientFactory.CreateArmClient<AzureRestClient>(context, AzureEnvironment.Endpoint.ResourceManager);
+                }
+
+                return _client;
+            }
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            this.ValidateParameters();
+
+            context = DefaultContext;
+            AzureOperationResponse<string> response;
+
+            if (!this.IsParameterBound(c => c.Path))
+            {
+                this.Path = Utils.ConstructPath(this.IsParameterBound(c => c.SubscriptionId) ? this.SubscriptionId : context.Subscription.Id, this.ResourceGroupName, this.ResourceProviderName, this.ResourceType, this.Name);
+            }
+
+            switch (this.Method)
+            {
+                case "GET":
+                    response = ServiceClient
+                    .Operations
+                    .GetResourceWithFullResponse(this.Path, this.ApiVersion);
+                    break;
+                case "POST":
+                    response = ServiceClient
+                    .Operations
+                    .PostResourceWithFullResponse(this.Path, this.ApiVersion, this.Payload);
+                    break;
+                case "PUT":
+                    response = ServiceClient
+                    .Operations
+                    .PutResourceWithFullResponse(this.Path, this.ApiVersion, this.Payload);
+                    break;
+                case "PATCH":
+                    response = ServiceClient
+                    .Operations
+                    .PatchResourceWithFullResponse(this.Path, this.ApiVersion, this.Payload);
+                    break;
+                case "DELETE":
+                    response = ServiceClient
+                    .Operations
+                    .DeleteResourceWithFullResponse(this.Path, this.ApiVersion);
+                    break;
+                default:
+                    throw new PSArgumentException("Invalid HTTP Method");
+            }
+
+            WriteObject(new PSHttpResponse(response));
+        }
+
+        public void ValidateParameters()
+        {
+            if (this.IsParameterBound(c => this.ResourceType) && !this.IsParameterBound(c => this.Name))
+            {
+                throw new PSArgumentException("number of resource types and resource names must be the same");
```

Comment:
I remember you have no plan to expose type and name pairs. If you still want to use it, it seems you need to add an additional parameter set to indicate resource type and name both need to be shown up. Then linke 152-160 are not required.